### PR TITLE
Feat/add i18n pages

### PR DESCRIPTION
**Pull request by Rob117**

### CODE REVIEW

File: @types/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export type SupportedLocales = 'en' | 'ja'
```
Comment (risafj) [inline @types/types.d.ts:1]: Let's make this be tracked by TS so semicolon checks are done

Comment (risafj) [reply @types/types.d.ts:1]: Also rename to index.d.ts

Comment (risafj) [reply @types/types.d.ts:1]: `SupportedLocale`

File: components/templates/homeTemplate.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Button, TextField } from '@mui/material';
```
Comment (risafj) [inline components/templates/homeTemplate.tsx:1]: Capital H for filename

File: components/templates/homeTemplate.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <main className="flex flex-col pt-10 px-4 justify-start">
+        <h1 className="text-5xl mb-auto">
+          NPM
+        </h1>
+
+        <p className="text-3xl py-20">
+          Premium { props.locale === 'en' ? 'mushrooms' : '椎茸' } at your doorstep, every month.
```
Comment (risafj) [inline components/templates/homeTemplate.tsx:21]: Slick

File: components/templates/loginTemplate.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
```
Comment (risafj) [inline components/templates/loginTemplate.tsx:1]: Capital for filename

File: pages/[locale]/index.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import { SupportedLocales } from '../../@types/types';
+import { HomeTemplate } from '../../components/templates/homeTemplate';
+import { createNonEnglishPaths } from '../../supportedLocales';
+
+const HomePage: NextPage<{locale: SupportedLocales}> = (props) => (<HomeTemplate { ...props } />);
+
+export const getStaticPaths: GetStaticPaths = () => createNonEnglishPaths();
+
+// pass the params from the path (mainly, the value of [locale]) to the component
+export const getStaticProps: GetStaticProps = async ({ params }) => {
```
Comment (risafj) [inline pages/[locale]/index.tsx:13]: async doesn't seem necessary?

File: pages/[locale]/index.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  return {
+    props: { ...params },
```
Comment (risafj) [inline pages/[locale]/index.tsx:15]: props: params

Comment (risafj) [reply pages/[locale]/index.tsx:15]: Try making one line function

**Review by risafj — CHANGES_REQUESTED** [review]
This is a request for changes

File: pages/[locale]/index.tsx (hunk 6 above)
Comment (Rob117) [reply pages/[locale]/index.tsx:15]: Typescript angry if no `...`

File: .eslintrc.json [hunk 7]
```diff
@@ -55,9 +55,10 @@
         "maxBOF": 0
       }
     ],
-    "semi": [
+    "semi": "off",
+    "@typescript-eslint/semi": [
       "error",
       "always"
     ]
   }
```
Comment (risafj) [inline .eslintrc.json:63]: Noice

**Review by risafj — APPROVED** [review]

**Pull request merged**